libraries/abstractions/common_io/test/test_iot_spi.c

```diff
@@ -1085,18 +976,101 @@ TEST( TEST_IOT_SPI, AFQP_IotSPI_TransferAsyncAssisted )
         #endif
     }
 
-    /* Append read bytes to string. */
-    for( i = 0, len = sizeof( ucRxBuf ); i < len; i++ )
+    prvAppendToMessage( &msgOffset, ucRxBuf, sizeof( ucRxBuf ) );
+    prvOutputMessage();
+}
+
+/**
+ * @brief Assisted Test Function to test spi write async
+ *
+ */
+TEST( TEST_IOT_SPI, AFQP_IotSPI_WriteAsyncAssisted )
+{
+    IotSPIHandle_t xSPIHandle;
+    IotSPIMasterConfig_t xOrigConfig, xTestConfig;
+    int32_t lRetVal;
+    uint8_t ucTxBuf[ 16 ] = { 0 };
+    char cMsg[ 50 ] = { 0 };
+    size_t xBytesTx;
+    size_t i = 0;
+    size_t msgOffset = 0;
+
+    srand( xTaskGetTickCount() );
+
+    /* Generate random tx bytes. */
+    for( ; i < sizeof( ucTxBuf ); i++ )
+    {
+        ucTxBuf[ i ] = ( uint8_t ) rand();
+    }
+
+    prvAppendToMessage( &msgOffset, ucTxBuf, sizeof( ucTxBuf ) );
+
+    /* If unit test and assisted test have different spi slave, make sure select slave function in
+     * ll hal is defined and select the slave for assisted slave. */
+    if( ulAssistedTestIotSpiSlave != ultestIotSpiSlave )
+    {
+        #ifdef IOT_TEST_COMMON_IO_SPI_SLAVE_SELECT_SUPPORTED
+            lRetVal = iot_spi_select_slave( ulAssistedTestIotSpiInstance, ulAssistedTestIotSpiSlave );
+            TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
+        #else
+            TEST_ASSERT_MESSAGE( 0, "Assisted test has a different salve, but slave select is not supported." );
+        #endif
+    }
+
+    /* Open SPI handle */
+    xSPIHandle = iot_spi_open( ulAssistedTestIotSpiInstance );
+    TEST_ASSERT_NOT_EQUAL( NULL, xSPIHandle );
+
+    /* save original configuration */
+    lRetVal = iot_spi_ioctl( xSPIHandle, eSPIGetMasterConfig, &xOrigConfig );
+    TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
+
+    if( TEST_PROTECT() )
+    {
+        /* configure bus */
+        xTestConfig.ulFreq = ultestIotSPIFrequency;
+        xTestConfig.eMode = xtestIotSPIDefaultConfigMode;
+        xTestConfig.eSetBitOrder = xtestIotSPIDefaultconfigBitOrder;
+        xTestConfig.ucDummyValue = ultestIotSPIDummyValue;
+
+        lRetVal = iot_spi_ioctl( xSPIHandle, eSPISetMasterConfig, &xTestConfig );
+        TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
+
+        /* Set a callback for async call */
+        iot_spi_set_callback( xSPIHandle, prvSpiAsyncCallback, NULL );
+
+        /* Make async write call */
+        lRetVal = iot_spi_write_async( xSPIHandle, ucTxBuf, sizeof( ucTxBuf ) );
+        TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
+
+        /*Wait for the callback. */
+        lRetVal = xSemaphoreTake( xtestIotSPISemaphore, testIotSPI_DEFAULT_SEMAPHORE_DELAY );
+        TEST_ASSERT_EQUAL( pdTRUE, lRetVal );
+
+        lRetVal = iot_spi_ioctl( xSPIHandle, eSPIGetTxNoOfbytes, &xBytesTx );
+        TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
+        TEST_ASSERT_EQUAL( xBytesTx, 4 );
+    }
+
+    /* restore original configuration */
```

Comment:
Can you please use complete sentences in comments?

Reply:
ok